pydantic/fields.py

```diff
@@ -784,6 +791,7 @@ def Field(
     description: str | None = _Unset,
     examples: list[Any] | None = _Unset,
     exclude: bool | None = _Unset,
+    exclude_if: Callable[[Any], bool] | None = _Unset,
```

Comment:
For this overload (`default` set) and the one for `default_factory` set, we can reuse the type variable to get proper type checking:

```python
exclude_if: Callable[[_T], bool] = _Unset
```